test/language/expressions/object/method-definition/generator-param-init-yield.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    When the `yield` keyword occurs within the Initializer of a
+    SingleNameBinding witihn the FormalParameters of a GeneratorMethod, it
+    behaves as a YieldExpression.
+es6id: 14.4
+features: [generators]
+flags: [noStrict]
+---*/
+
+var yield = 'defaultViaIdentifier';
+var obj;
+var iter = (function*() {
+  obj = {
+    *method(x = yield) {
+      return x;
+    }
+  };
+}());
+
+iter.next();
+
+iter.next('defaultViaExpression');
+
+assert.sameValue(obj.method().next(), 'defaultViaExpression');
```

Comment:
There is definitely a mistake here, but I think the correction looks more like:

```diff
 iter.next('defaultViaExpression');
 
-assert.sameValue(obj.method().next(), 'defaultViaExpression');
+assert.sameValue(obj.method().next().value, 'defaultViaExpression');
```

Here's my reasoning:

The first call to `next` is necessary to start the generator and is not associated with any `yield` expression. The next call to `next` binds the value of the `yield` expression in the method's initializer and completes the definition of `obj`. At this point, `obj.method` is equivalent to `(function*(x = 'defaultViaExpression') { return x; })`, so `obj.method()` will create an iterator that returns `'defaultViaExpression'`.

This is one of the more convoluted tests I've written so far, though, so the likelyhood that I still have it wrong is high